src/main/java/com/google/cloud/tools/appengine/configuration/AppYamlProjectStageConfiguration.java

```diff
@@ -89,14 +102,21 @@ public static Builder builder(Path appEngineDirectory, Path artifact, Path stagi
     private Path artifact;
     private Path stagingDirectory;
 
-    Builder(Path appEngineDirectory, Path artifact, Path stagingDirectory) {
-      Preconditions.checkNotNull(appEngineDirectory);
-      Preconditions.checkNotNull(artifact);
-      Preconditions.checkNotNull(stagingDirectory);
+    private Builder() {}
 
+    public Builder appEngineDirectory(Path appEngineDirectory) {
       this.appEngineDirectory = appEngineDirectory;
```

Comment:
I see that this method no longer does `Preconditions.checkNotNull`, but I this is fine, because `build()` will check these at the end of the day.